src/Synth/OscilGen.h

```diff
@@ -114,21 +114,50 @@ class OscilGen:public Presets
         /* Oscillator Frequencies -
          *  this is different than the harmonics set-up by the user,
          *  it may contains time-domain data if the antialiasing is turned off*/
-        fft_t *oscilFFTfreqs;
-
-        fft_t *pendingfreqs;
+        class deep_copying_fft_ptr
+        {
+            fft_t* ptr;
+            std::size_t size = 0;
+        public:
+            deep_copying_fft_ptr() = default;
+            void init_size(std::size_t newsize);
+            void reset(fft_t* new_ptr) { ptr = new_ptr; }
+            deep_copying_fft_ptr(const deep_copying_fft_ptr& other);
+            fft_t& operator[](std::size_t i);
+            const fft_t& operator[](std::size_t i) const;
+            operator fft_t*() { return ptr; }
+            operator const fft_t*() const { return ptr; }
+        } oscilFFTfreqs;
+
+        fft_t* pendingfreqs;
     private:
         //This array stores some temporary data and it has OSCIL_SIZE elements
-        float *tmpsmps;
-        fft_t *outoscilFFTfreqs;
-        float *cachedbasefunc;
+        std::vector<float> tmpsmps;
+        std::vector<fft_t> outoscilFFTfreqs;
+        std::vector<float> cachedbasefunc;
         bool cachedbasevalid;
 
         float hmag[MAX_AD_HARMONICS], hphase[MAX_AD_HARMONICS]; //the magnituides and the phases of the sine/nonsine harmonics
 
-        FFTwrapper *fft;
+        class fftWrapperContainer
+        {
+            FFTwrapper *fft;
+            bool allocatedByUs;
+        public:
+            //! Refering CTOR - OscilGen default ctor will use this
+            fftWrapperContainer(FFTwrapper* fft) : fft(fft), allocatedByUs(false) {}
+            //! Deep copying CTOR - OscilGen copy ctor will use this
+            fftWrapperContainer(const fftWrapperContainer& source);
+            ~fftWrapperContainer();
+            operator FFTwrapper*() { return fft; }
+            operator const FFTwrapper*() const { return fft; }
+            FFTwrapper* operator->() { return fft; }
+            const FFTwrapper* operator->() const { return fft; }
+        } fft;
```

Comment:
I'd be really wary about this functionality. We can't have full smart pointers due to the whole multi-threaded situation and this copy would allow for a "weak pointer", which is a complexity that can bite back in unexpected ways. Also, with regards to FFTW, there are modes which assume that the library knows the address of the pointer passed to it. I don't think we're currently using one, but we may end up doing so in the future and this would complicate things as the pointer address of some of the fft_t* references would change.